Set use_mirrors default to "no" for the pip module

TallGirlVanessa opened the pull request:
Also deprecate `use_mirrors`, if that's ok! I hope I'm not overstepping my bounds; it just seems like a good idea.

Anyway, PyPI moved to a CDN on 2013-05-26, so mirrors are now largely unnecessary and pip removed the functionality in 1.5. More importantly (and why I wrote this request): on 2014-02-15 the mirror directory was taken offline, so mirrors may not work anymore even for pip versions that support them. Somehow one of my machines was able to install things anyway (using pip 1.0), but another one of my machines hung indefinitely when trying to install a new version of a package.

mpdehaan commented:
What should be done here IMHO is to keep the argument to "eat" the parameter and remove the support for mirrors from the code.

TallGirlVanessa committed:
Remove use_mirrors from pip module

PyPI moved to a CDN on 2013-05-26, so mirrors are now largely
unnecessary and pip removed the functionality in 1.5. More importantly
(and why I wrote this request): on 2014-02-15 the mirror directory was
taken offline, so mirrors may not work anymore even for pip versions
that support them.

TallGirlVanessa committed:
pip module wasn't adding -e when using VCS

TallGirlVanessa commented:
I think I've fixed it. I also think I fixed an unrelated problem: it looks like -e is automatically added to extra_args when pip installing a VCS URL, but by that time extra_args has already been added to cmd and isn't read again.

mpdehaan commented:
I think we should also remove use_mirrors from the comments, as it's not deprecated, it's gone.

Can you update this? 

Thanks!

TallGirlVanessa commented:
Yeah, I was wondering while I was writing it if so much documentation for a feature you guys were removing was a good idea :smiley: I've removed it.

mpdehaan commented:
Merged, thanks!

TallGirlVanessa commented:
NP, thanks for the feedback!

dhozac commented:
Adding -e breaks existing deployment playbooks that expect it to not be set, especially as there is no way to turn it off.

mpdehaan commented:
Hi @dhozac !!!

TallGirlVanessa commented:
Sorry about that; I noticed the examples for the module say VCS URLs automatically get -e and that somewhere along the line that logic broke, so I "fixed" it. But yeah, not being able to turn it off kinda sucks.

What's the protocol for when a merged PR has a bug? Should I open a new one?
